wallace/custom.py

```diff
@@ -420,31 +410,17 @@ def vector():
     # load the experiment
     exp = experiment(session)
 
-    # get the participant_id
-    try:
-        participant_id = request.values["participant_id"]
-        key = participant_id[0:5]
-    except:
-        exp.log("Error: /vector request, participant_id not specified")
-        page = error_page(error_type="/vector, participant_id not specified")
-        js = dumps({"status": "error", "html": page})
-        return Response(js, status=403, mimetype='application/json')
+    # get the key
+    key = participant_id[0:5]
 
-    # get the node_id
-    try:
-        node_id = request.values["node_id"]
-        if not node_id.isdigit():
-            exp.log(
-                "Error: /vector request, non-numeric node_id: {}"
-                .format(node_id), key)
-            page = error_page(error_type="/vector, non-numeric node_id")
-            js = dumps({"status": "error", "html": page})
-            return Response(js, status=403, mimetype='application/json')
-    except:
-        exp.log("Error: /vector request, node_id not specified", key)
-        page = error_page(error_type="/vector, node_id not specified")
+    # check the node_id
+    if not node_id.isdigit():
+        exp.log(
+            "Error: /vector request, non-numeric node_id: {}"
+            .format(node_id), key)
+        page = error_page(error_type="/vector, non-numeric node_id")
```

Comment:
If it is always an error for node_id to be non-numeric, you can again use flask's routing power to help you here.  For example consider this route (from the flask documentation):

```python
@app.route('/post/<int:post_id>')
```
Here only requests with integer-safe values for post_id will be matched and passed on.  So you can be sure that requests for paths such as `/post/abc` will never match, but `/post/237856` will match.  Additionally, the `post_id` received as an input to the decorated view function will already be cast to an integer, which saves a bit of effort in type conversion.